oml/utils/misc_torch.py

```diff
@@ -58,6 +59,41 @@ def _check_is_sequence(val: Any) -> bool:
         return False
 
 
+def label_smoothing(
+    y: torch.Tensor,
+    num_classes: int,
+    label_smoothing: float = 0.2,
+    label2category: Optional[torch.Tensor] = None,
+) -> torch.Tensor:
+    """
+    This function is doing `label smoothing <https://arxiv.org/pdf/1512.00567v3.pdf>`_.
+    You can also use modified version, where the label is smoothed only for the same category.
+    To use this, you should provide the ``label2category`` argument: vector, for which i-th entry
```

Comment:
I don't like that we have the same name (`label2category`) for a dict and tensor (yes, in different functions, but still, they are related) 
Perhaps, we can name it as `categories`?